### ISSUE TITLE
Slices backed by AWS Athena load sequentially, not in parallel.

### ISSUE THREAD

**Issue by sjmeyer** [apache/incubator-superset#5279]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version

Versions: 0.22.1, 0.23, 0.25.6

### Expected results

Slices should load simultaneously. Athena logs should also show queries to be called at the same time. 

### Actual results

Slices load one after the other, making for poor user experience. The Athena logs also shows a query is called only once the previous one finishes.

Athena log timestamps:

    2018/06/25 13:53:43 UTC+-2 Query...
    2018/06/25 13:53:41 UTC+-2 Query...
    2018/06/25 13:53:37 UTC+-2 Query...
    2018/06/25 13:53:36 UTC+-2 Query...
    2018/06/25 13:53:36 UTC+-2 Query...
    2018/06/25 13:53:34 UTC+-2 Query...
    2018/06/25 13:53:31 UTC+-2 Query...

### Steps to reproduce

- Build a dashboard with multiple slices, each built on it's own Athena query.
- Use the below gunicorn command line arguments:

   GUNICORN_CMD_ARGS "-w 2 -k gevent --threads 12 --timeout 120 -b  0.0.0.0:8088 --limit-request-line 0  --limit-request-field_size 0 --proxy-protocol true --proxy-allow-from * --forwarded-allow-ips * --keep-alive 75"

**Comment by sjmeyer** [comment]
I changed my configuration to what you suggested. I tried both with 8 workers and 16 workers and no --threads argument. Neither of these worked.

I can still see queries running one after the other in the Athena logs.

**Comment by mistercrunch** [comment]
Looks like Athena is the thing limiting concurrency somehow.

**Comment by sjmeyer** [comment]
@mistercrunch I'm busy doing some more digging around and found something interesting. I am logging out the timestamp when the Ajax from a slice hits the explore_json endpoint, and I'm logging out when the query() method is called to execute the query. Even though I have 16 workers, only 4 Ajax calls are handled at first. After that some are staggered and some are handled two by two.

When it comes to executing the queries, they are being run in sequence. It is clear from the logs that one is waiting for another to finish before running. Is it possible that the issue is with the JDBC driver or the way in which that connection is made?

Also, any idea why only 4 Ajax requests would be handled even when 16 workers are used?

**Comment by mistercrunch** [comment]
What's your `-w` argument?

**Comment by mistercrunch** [comment]
The Celery workers are mostly there to deal with long running queries and heavy workloads.

`create_engine` being called for each query is right, we do not use connection pooling for external databases, only for the connections to the metadata database.

**Comment by sjmeyer** [comment]
@mistercrunch It seems like the issue is the running of the JDBC driver jar. A new JVM gets created for each connection, instead of threads being added to one running JVM. I think it has something to do with SQLAlchemy wanting one thread per process.

**Comment by LeoDiep** [comment]
Hi @sjmeyer Is there a way to config PyAthena to send multiple queries to Athena so that Superset can have better performance on dashboard which have more than 10 slices. Everytime I apply a filters, it takes 45s to fully load visualization for all slices

**Comment by Abhijith-Nagaraja** [comment]
Hi @sjmeyer @LeoDiep: Is this issue resolved? same issue I am experiencing even after enabling Global Async queries. Opened a new ticket as this ticket is marked as completed
https://github.com/apache/superset/issues/25827

**Comment by LeoDiep** [comment]
> Hi @sjmeyer @LeoDiep: Is this issue resolved? same issue I am experiencing even after enabling Global Async queries. Opened a new ticket as this ticket is marked as completed #25827

Hi, I solved my problem by increasing the number of gunicorn workers from 1 (default) to 6 workers, now 45s decrease to 6s

**Comment by Abhijith-Nagaraja** [comment]
@LeoDiep : That's what I ended up doing. Thanks for suggesting it. Although I don't know why async queries are not working. I hope someone from superset team reviews by request in https://github.com/apache/superset/issues/25827

**Comment by LeoDiep** [comment]
> @LeoDiep : That's what I ended up doing. Thanks for suggesting it. Although I don't know why async queries are not working. I hope someone from superset team reviews by request in #25827

I guess async queries are only applied to long response time queries (more than 10s each) so that the sql lab doesn't get stuck when querying other queries
